Added documentation for the new directory layout

tsg opened the pull request:
This adds the doc changes for #1371. 

@dedemorton can you have a look if I gave enough details, now? For editing I think it's easier to merge it and edit it in future PRs, right?

dedemorton commented on libbeat/docs/shared-path-config.asciidoc:
```diff
@@ -0,0 +1,96 @@
+===== data
+
+The data path for a {beatname_uc} installation is the default base path for all
+the files in which {beatname_uc} needs to store its files. If not set via a CLI
```
I think yo mean to say "for all the folders in which...." (instead of "for all the files in which").

This description still seems a bit vague. It would help if you could say a bit more about how the files are used. Maybe providing an example or two would help.

dedemorton replied:
Oh you should probably say directories instead of folders here, too.

tsg replied:
Damn, I meant "store its data", not files. The reason I'm being vague is that this file is included for all beats, but not all beats put stuff in the data path. It's only Filebeat and Winlogbeat, which store their registry file. Topbeat and Packetbeat don't put anything in it for now, but that will probably change soon.  That's why I'm not sure how to make it more concrete.  Maybe we should just explain it like that (Filebeat and Winlogbeat store the registry file, Topbeat and Packetbeat currently don't use this setting).

dedemorton commented on libbeat/docs/shared-path-config.asciidoc:
```diff
@@ -0,0 +1,96 @@
+=== Path
+
+The `path` section contains configuration options that define where the Beat
+is looking for certain types of files.
```
The statement "certain types of files" is a bit too vague. Maybe be specific about the kinds of data that the files contain or how they are used.

dedemorton commented on libbeat/docs/shared-path-config.asciidoc:
```diff
@@ -0,0 +1,96 @@
+
+===== logs
+
+The data path for a {beatname_uc} installation is the default base path for all
```
The log path

dedemorton commented on libbeat/docs/shared-path-config.asciidoc:
```diff
@@ -0,0 +1,96 @@
+===== logs
+
+The data path for a {beatname_uc} installation is the default base path for all
+the files in which {beatname_uc} needs to store its files. If not set via a CLI
```
Change "files" to "folders"

tsg replied:
Oops, that was totally wrong due to a copy and paste error. I'll fix it.

dedemorton commented:
@tsg I've added a couple of comments where I think you need to say a bit more.

tsg commented:
I fixed some things and added some examples at the beginning of the configuration page, hopefully that helps a little. It's difficult to be specific because some things depend from beat to beat, but these files are common.

dedemorton commented:
@tsg: LGTM. I'd also like to take a pass at editing this later, so please open a doc issue for that.

andrewkroh commented:
LGTM
